### ISSUE TITLE
Method one() is not working as expected

### ISSUE THREAD

**Issue by redaxmedia** [kupriyanenko/jbone#45]
The method <code>one()</code> from <code>event.js</code> worked fine in 1.0.22 but fails now.

 I don't have the time for reengineering your changes for the <code>on()</code> and <code>Event</code>. Sorry, but your code lacks of documentation and is not that easy to understand for me. 

Nothing is lost because I can provide a modified test that is working in 1.0.22 but fails in 1.0.25

<pre>
    it('one(event, callback)', function() {
        var div = jBone('&lt;div&gt;'), counter = 0, fn = function() {
            counter++;
        };

        div.one('click', fn);

        div.trigger('click').trigger('click');

        expect(counter).be.eql(1);

        var divs = jBone('&lt;input&gt;&lt;input&gt;');

        divs.one('foo', fn);

        divs.eq(0).trigger('foo');

        divs.one('foo', fn);

        divs.eq(1).trigger('foo');
        divs.eq(1).trigger('foo');

        expect(counter).be.eql(4);
    });
</pre>

The issue appears once the <code>one()</code> was fired and added again... it does not fire another time

**Comment by redaxmedia** [comment]
For those who run into the same issue, here is a tempfix branch that can be used temporary with bower:

<pre>
"dependencies": {
     "jbone": "https://github.com/redaxmedia/jbone.git#tempfix"
}

**Comment by kupriyanenko** [comment]
Thanks for founded bug. Fixed in [1.0.26](https://github.com/kupriyanenko/jbone/releases/tag/1.0.26)

**Comment by redaxmedia** [comment]
1. I cannot reproduce this with a unit test, but in my pageflip app the one() method is not working multiple times - only 1 to 2...

2. So I switched back to on() but then jBone suffers in performace like mentioned in another issue

3. I suggest to drop the cache improvements and switch event.js back to 1.0.22 - it just does not work

**Comment by redaxmedia** [comment]
Working on:

<pre>
    fn.on = function(event) {
        var args = arguments,
            length = this.length,
            i = 0,
            callback = slice.call(args, -1)[0],
            target, data, namespace, fn, events, eventType, expectedTarget, addListener;

        // .on('click', '.selector', function() {})
        if (args.length === 3 &amp;&amp; isString(args[1])) {
            target = args[1];
        }
        // .on('click', { key: value }, function() {})
        else if (args.length === 3 &amp;&amp; isObject(args[1])) {
            data = args[1];
        }
        // .on('click', '.selector', { key: value }, function() {})
        else if (args.length === 4) {
            target = args[1];
            data = args[2];
        }

        addListener = function(el) {
            jBone.setId(el);
            events = jBone.getData(el).events;
            event.split(&quot; &quot;).forEach(function(event) {
                var eventOptions = {};

                if (data) {
                    eventOptions.data = data;
                }

                eventType = event.split(&quot;.&quot;)[0];
                namespace = event.split(&quot;.&quot;).splice(1).join(&quot;.&quot;);
                events[eventType] = events[eventType] || [];

                fn = function(e) {
                    if (e.namespace &amp;&amp; e.namespace !== namespace) {
                        return;
                    }

                    expectedTarget = null;

                    if (!target) {
                        callback.call(el, new BoneEvent(e, eventOptions));
                    } else if (~jBone(el).find(target).indexOf(e.target) || (expectedTarget = jBone.contains(jBone(el).find(target), e.target))) {
                        expectedTarget = expectedTarget || e.target;
                        eventOptions.currentTarget = expectedTarget;

                        callback.call(expectedTarget, new BoneEvent(e, eventOptions));
                    }
                };

                events[eventType].push({
                    namespace: namespace,
                    fn: fn,
                    originfn: callback
                });

                el.addEventListener &amp;&amp; el.addEventListener(eventType, fn, false);
            });
        };

        for (; i &lt; length; i++) {
            addListener(this[i]);
        }

        return this;
    };
</pre>

Not working:

<pre>
        fn.one = function(event) {
        var args = arguments,
            i = 0,
            length = this.length,
            oneArgs = slice.call(args, 1, args.length - 1),
            callback = slice.call(args, -1)[0],
            addListener;

        addListener = function(el) {
            var $el = jBone(el);

            event.split(&quot; &quot;).forEach(function(event) {
                var fn = function(e) {
                    $el.off(event, fn);
                    callback.call(el, e);
                };

                $el.on.apply($el, [event].concat(oneArgs, fn));
            });
        };

        for (; i &lt; length; i++) {
            addListener(this[i]);
        }

        return this;
    };
</pre>

**Comment by kupriyanenko** [comment]
Can you share with me your code for pageflip, if I will reproduce it, I will fix it :)

**Comment by redaxmedia** [comment]
I cannot setup the pageflip online because it is still in development and needs an API to load the brochures... I cannot reproduce the problem in a jsfiddle, because the pageflip is a complex framework :-)

1. After navigate a Event <code>renderStart</code> get fired as the pageflip refreshed the GUI
1. A Method does look if the top or bottom is opened and fires the Event <code>slideTopOpen</code> only once

Object <code>jBone._cache.events</code> may cause the problem...

**Latest jBone - the overview fails to render after multiple navigate:**

![Fixed jBone](http://i.snag.gy/hF1bA.jpg)

**Fixed jBone - the overview does render perfect:**

![Latest jBone](http://i.snag.gy/0yddi.jpg)

**Comment by kupriyanenko** [comment]
Please try version [1.0.27](https://github.com/kupriyanenko/jbone/releases/tag/1.0.27), I think bug should be fixed.

**Comment by kupriyanenko** [comment]
And I hope tests for your case: [add test for handle multiple on()/off()](https://github.com/kupriyanenko/jbone/commit/f42552fbe47ccdfce88c444cb38aa6effed118aa)

**Comment by redaxmedia** [comment]
Works for me, thank you so much!
